ietfdata/datatracker.py

```diff
@@ -1261,6 +1295,64 @@ def meeting_types(self) -> Iterator[MeetingType]:
         """
         return self._retrieve_multi("/api/v1/name/meetingtypename/", MeetingType)
 
+# =================================================================================================================================
+#   https://datatracker.ietf.org/api/v1/doc/relateddocument/?source=...      - documents that source draft relates to (references, replaces, etc)
+#   https://datatracker.ietf.org/api/v1/doc/relateddocument/?target=...      - documents that relate to target draft
+
+    def related_documents(self, 
+            source               : Optional[Document]         = None, 
+            target               : Optional[Document]         = None, 
+            relationship_type    : Optional[RelationshipType] = None) -> Iterator[RelatedDocument]:
+
+        url = "/api/v1/doc/relateddocument/"
+        if source is not None and target is not None and relationship_type is not None:
+            url = url + "?source=" + source + "&target=" + target + "&relationship=" + relationship_type
+        elif source is not None and target is not None:
+            url = url + "?source=" + source + "&target=" + target
+        elif source is not None and relationship_type is not None:
+            url = url + "?source=" + source + "&relationship=" + relationship_type
+        elif target is not None and relationship_type is not None:
+            url = url + "?target=" + target + "&relationship=" + relationship_type
+        elif target is not None:
+            url = url + "?target=" + target
+        elif source is not None:
+            url = url + "?source=" + source
+        elif relationship_type is not None:
+            url = url + "?relationship=" + relationship_type
+        return self._retrieve_multi(url, RelatedDocument)
+    
+    
+    def relationship_type(self, relationship_type: str) -> Optional[RelationshipType]:
```

Comment:
Why does this take a slug rather than a relationship_type_uri?

Reply:
Now takes a relationship type uri